src/schoology-extractor/schoology_extractor/api/request_client.py

```diff
@@ -256,33 +256,26 @@ def get_section_by_course_ids(
         return sections
 
     def get_submissions_by_section_id_and_grade_item_id(
-        self, section_id: str, grade_item_id: str, page_size: int = DEFAULT_PAGE_SIZE
+        self, section_id: int, grade_item_id: int, page_size: int = DEFAULT_PAGE_SIZE
     ) -> PaginatedResult:
         """
+        Retrieves submissions for assignments or discussions in a section.
+
         Parameters
         ----------
-        section_id : str
-            The id of the section.
-        grade_item_id : str
-            Grade item id.
+        section_id : int
+            The id of the section
+        grade_item_id : int
+            Grade Item Id, which can either be an Assignment ID or a Discussion ID
         page_size : int
-            Number of items per page.
+            Number of items per page
 
         Returns
         -------
         PaginatedResult
             A parsed response from the server
 
         """
-        assert isinstance(
-            section_id, str
-        ), "Argument `section_id` should be of type `str`."
-        assert isinstance(
-            page_size, int
-        ), "Argument `page_size` should be of type `int`."
-        assert isinstance(
-            grade_item_id, str
-        ), "Argument `grade_item_id` should be of type `str`."
```

Comment:
Now that I'm trusting the static type checking process, I no longer feel bound to have these assert statements all over the place. Opportunistically removed these to simplify the code.